protocol/common.go

```diff
@@ -0,0 +1,18 @@
+package protocol
+
+import (
+	"encoding/hex"
+	"fmt"
+
+	"github.com/coniks-sys/coniks-go/crypto"
+)
+
+// ComputeDirectoryIdentity returns the hash of
+// the directory's initial STR as a string.
+// It panics if the STR isn't an initial STR (i.e. str.Epoch != 0).
+func ComputeDirectoryIdentity(str *DirSTR) string {
+	if str.Epoch != 0 {
+		panic(fmt.Sprintf("[coniks] Expect epoch 0, got %x", str.Epoch))
+	}
+	return hex.EncodeToString(crypto.Digest(str.Signature))
```

Comment:
Why the hex encoding here?  We do `bytes.Equal` with the `str.PreviousSTRHash` in other places.  Maybe it'd be better to keep all these `crypto.Digest(Signature)` operations the same.